scripts/rs/main.rs

```diff
@@ -0,0 +1,159 @@
+use std::process::Command;
+use std::thread::sleep;
+use std::time::Duration;
+use redis::{Commands, Connection, Client};
+use regex::Regex;
+use sysinfo::{System, Process};
+use chrono::Utc;
+use std::fs;
+use std::collections::HashMap;
+
+
+const REDIS_URL: &str = "redis://127.0.0.1/";
+const UPDATE_INTERVAL: u64 = 120; // 2 minutes
+
+
+fn run_command(cmd: &[&str]) -> Option<String> {
+    let output = Command::new(cmd[0]).args(&cmd[1..]).output().ok()?;
+    if output.status.success() {
+        Some(String::from_utf8_lossy(&output.stdout).to_string())
+    } else {
+        eprintln!("Error running command: {:?}", cmd);
+        None
+    }
+}
+
+fn convert_to_bytes(value: &str) -> u64 {
+    let re = Regex::new(r"(\d+\.?\d*)([a-zA-Z]+)").unwrap();
```

Comment:
This recompiles the regex on every call. This is bad practice, though admittedly if it's just once every 2 minutes it won't matter. This code will cache it for you.
```rust
fn convert_to_bytes(value: &str) -> u64 {
    static RE: Regex = LazyLock::new(|| Regex::new(r"(\d+\.?\d*)([a-zA-Z]+)").unwrap());
    if let Some(caps) = RE.captures(value) {
    ...
```